pallets/runtime/tests/src/external_agents_test.rs

```diff
@@ -247,3 +250,69 @@ fn add_works() {
         assert_noop!(accept(dave, id), Error::NumFullAgentsOverflow);
     });
 }
+
+fn create_asset(ticker: &[u8], owner: User) -> Ticker {
+    let (ticker, token) = token(ticker, owner.did);
+    assert_ok!(basic_asset(owner, ticker, &token));
+    ticker
+}
+
+#[test]
+fn agent_of_mapping_works() {
+    ExtBuilder::default().build().execute_with(|| {
+        let owner = User::new(AccountKeyring::Alice);
+        let bob = User::new(AccountKeyring::Bob);
+        let charlie = User::new(AccountKeyring::Charlie);
+        let dave = User::new(AccountKeyring::Dave);
+        let tickers = (b'A'..b'Z')
+            .into_iter()
+            .map(|ticker| create_asset(&[ticker], owner))
+            .collect::<Vec<_>>();
+
+        let add = |ticker, from: User, to: User, group| {
+            let data = AuthorizationData::BecomeAgent(ticker, group);
+            let sig = Signatory::Identity(to.did);
+            assert_ok!(Id::accept_authorization(
+                to.origin(),
+                Id::add_auth(from.did, sig, data, None)
+            ));
+        };
+        let check = |user: User| {
+            let mut agents = AgentOf::iter_prefix(user.did)
+                .map(|(ticker, _)| ticker)
+                .collect::<Vec<_>>();
+            agents.sort();
```

Comment:
You can also test this against `GroupOfAgent::iter().filter(|(_, d, _)| d == user.did).map(|(t, ..)| t).collect()` and ensure consistency, which is really the crucial property.